File: flower/src/main/java/com/kharitonov/flower/contracts/ApiRouter.java
package com.kharitonov.flower.contracts;

public class ApiRouter {

    public static class AuthRouter {
        public static final String BASE_URL = "api/v1/auth";
        public static final String LOGIN = "/login";
        public static final String TOKEN = "/token";
        public static final String REFRESH = "/refresh";
    }

    public static class UserRouter {
        public static final String BASE_URL = "api/v1/user";
        public static final String FIND_BY_ID = "/{id}";
        public static final String UPDATE = "/{id}";
        public static final String DELETE_BY_ID = "/{id}";
    }

    public static class FlowerRouter {
        public static final String BASE_URL = "api/v1/flower";
        public static final String FIND_BY_ID = "/{id}";
        public static final String UPDATE = "/{id}";
        public static final String DELETE_BY_ID = "/{id}";
    }

    public static class BasketFlower {
        public static final String BASE_URL = "api/v1/flower";
        public static final String FIND_BY_ID = "/{id}";
        public static final String UPDATE = "/{id}";
        public static final String ADD_FLOWER = "/toBasketId/flowerId";
        public static final String DELETE_BY_ID = "/{id}";
    }

}

File: src/main/java/by/hghgrtut/shop/contracts/ApiRouter.kt
package by.hghgrtut.shop.contracts

class ApiRouter {
    
    companion object {

        private const val API_BASE_URL = "api/v1"
    }

    object AuthRouter {
        const val BASE_URL = "${API_BASE_URL}auth"
        const val LOGIN = "/login"
        const val TOKEN = "/token"
        const val REFRESH = "/refresh"
    }

    object UserRouter {
        const val BASE_URL = "${API_BASE_URL}user"
        const val FIND_BY_ID = "/{id}"
        const val UPDATE = "/{id}"
        const val DELETE_BY_ID = "/{id}"
    }

    object ItemRouter {
        const val BASE_URL = "${API_BASE_URL}item"
        const val FIND_BY_ID = "/{id}"
        const val UPDATE = "/{id}"
        const val DELETE_BY_ID = "/{id}"
    }

    object BasketItem {
        const val BASE_URL = "${API_BASE_URL}item"
        const val FIND_BY_ID = "/{id}"
        const val UPDATE = "/{id}"
        const val ADD_FLOWER = "/basket/itemId"
        const val DELETE_BY_ID = "/{id}"
    }
}